app/javascript/src/components/Dashboard/Contacts/index.jsx

```diff
@@ -1,25 +1,81 @@
 import React, { useState, useEffect } from "react";
 
 import EmptyListImage from "images/EmptyList";
-import { Button } from "neetoui";
+import { Button, Toastr } from "neetoui";
 import { Container, Header } from "neetoui/layouts";
 
 import EmptyState from "components/Common/EmptyState";
 import SideMenuBar from "components/Common/SideMenuBar";
 import { CONTACTS } from "components/constants";
 
-import { CONTACTS_MENU_ITEMS } from "./constants";
+import {
+  CONTACTS_FORM_INITIAL_FORM_VALUES,
+  CONTACTS_MENU_ITEMS,
+} from "./constants";
+import CreateOrEditPane from "./Pane/CreateOrEdit";
 import Table from "./Table";
+import { formatContactDataForEdit, formatContactDataForSave } from "./utils";
 
 const Contacts = () => {
   const [contacts, setContacts] = useState([]);
   const [searchTerm, setSearchTerm] = useState("");
+  const [selectedContactForEditOrDelete, setSelectedContactForEditOrDelete] =
+    useState({});
   const [showSideMenuBar, setShowSideMenuBar] = useState(true);
+  const [showCreateOrEditPane, setShowCreateOrEditPane] = useState(false);
+  const [isEditContact, setIsEditContact] = useState(false);
 
   useEffect(() => {
     setContacts(CONTACTS);
   }, []);
 
+  const resetStates = () => {
+    setShowCreateOrEditPane(false);
+    setSelectedContactForEditOrDelete({});
+    setIsEditContact(false);
+    setShowSideMenuBar(false);
+  };
+
+  const handleCreate = () => {
+    setIsEditContact(false);
+    setSelectedContactForEditOrDelete({});
+    setShowCreateOrEditPane(true);
+  };
+
+  const handleEdit = contact => {
+    setIsEditContact(true);
+    setSelectedContactForEditOrDelete(formatContactDataForEdit(contact));
+    setShowCreateOrEditPane(true);
+  };
+
+  const handleSubmit = ({ contact, isEdit }) => {
+    if (isEdit) {
+      const indexOfSelectedContact = contacts.findIndex(
+        ({ id }) => id === contact.id
+      );
+      if (indexOfSelectedContact === -1) return;
+
+      contacts[indexOfSelectedContact] = formatContactDataForSave(contact);
+    } else {
+      const sortedContactsById = contacts.sort((a, b) => a.id - b.id);
+
+      const nextId =
+        sortedContactsById.length === 0
+          ? 1
+          : sortedContactsById[sortedContactsById.length - 1].id + 1;
+      contact.id = nextId;
+      contacts.push(formatContactDataForSave(contact));
```

Comment:
This suggests few things to me:
- You haven't understood how React state hooks work, and the purpose of a state setter
- You haven't understood what mutation is in JavaScript and why it should be avoided
- You are not paying very close attention to the code that you are writing

I want you to carefully go through this line of code and put up a video and point out to me what the issue is over here and also point out what's the right way to have written this code.

You should do this with the highest priority. You can pause all other work that you are doing and put up this video.